pkg/services/comments/handlers.go

```diff
@@ -0,0 +1,157 @@
+package comments
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sort"
+
+	"github.com/grafana/grafana/pkg/api/dtos"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/comments/commentmodel"
+)
+
+func commentsToDto(items []*commentmodel.Comment, userMap map[int64]*models.UserSearchHitDTO) []*commentmodel.CommentDto {
+	result := make([]*commentmodel.CommentDto, 0, len(items))
+	for _, m := range items {
+		result = append(result, commentToDto(m, userMap))
+	}
+	return result
+}
+
+func commentToDto(comment *commentmodel.Comment, userMap map[int64]*models.UserSearchHitDTO) *commentmodel.CommentDto {
+	var u *commentmodel.CommentUser
+	if comment.UserId > 0 {
+		user, ok := userMap[comment.UserId]
+		if !ok {
+			// TODO: handle this gracefully?
+			u = &commentmodel.CommentUser{
+				Id: comment.UserId,
+			}
+		} else {
+			u = &commentmodel.CommentUser{
+				Id:        user.Id,
+				Name:      user.Name,
+				Login:     user.Login,
+				Email:     user.Email,
+				AvatarUrl: dtos.GetGravatarUrl(user.Email),
+			}
+		}
+	}
+	return comment.ToDTO(u)
+}
+
+type UserIDFilter struct {
+	userIDs []int64
+}
+
+func NewIDFilter(userIDs []int64) models.Filter {
+	return &UserIDFilter{userIDs: userIDs}
+}
+
+func (a *UserIDFilter) WhereCondition() *models.WhereCondition {
+	return nil
+}
+
+func (a *UserIDFilter) JoinCondition() *models.JoinCondition {
+	return nil
+}
+
+func (a *UserIDFilter) InCondition() *models.InCondition {
+	return &models.InCondition{
+		Condition: "u.id",
+		Params:    a.userIDs,
+	}
+}
+
+type GetCmd struct {
+	ObjectType string `json:"objectType"`
+	ObjectID   string `json:"objectId"`
+	Limit      uint   `json:"limit"`
+	BeforeId   int64  `json:"beforeId"`
+}
+
+type CreateCmd struct {
+	ObjectType string `json:"objectType"`
+	ObjectID   string `json:"objectId"`
+	Content    string `json:"content"`
+}
+
+var ErrPermissionDenied = errors.New("permission denied")
+
+func (s *Service) Create(ctx context.Context, orgID int64, signedInUser *models.SignedInUser, cmd CreateCmd) (*commentmodel.CommentDto, error) {
+	ok, err := s.permissions.CheckWritePermissions(ctx, orgID, signedInUser, cmd.ObjectType, cmd.ObjectID)
+	if err != nil {
+		return nil, err
+	}
+	if !ok {
+		return nil, ErrPermissionDenied
+	}
+
+	userMap := map[int64]*models.UserSearchHitDTO{}
+	if signedInUser.UserId > 0 {
+		q := &models.SearchUsersQuery{Query: "", Filters: []models.Filter{NewIDFilter([]int64{signedInUser.UserId})}, Page: 0, Limit: 1}
+		if err := s.sqlStore.SearchUsers(ctx, q); err != nil {
+			return nil, err
+		}
+		for _, u := range q.Result.Users {
+			userMap[u.Id] = u
+		}
```

Comment:
why are you searching for the signed-in user?

Reply:
@FZambia we get avatarUrl in index.go from just SignInUser 

dtos.GetGravatarUrl(c.Email) 

https://github.com/grafana/grafana/blob/main/pkg/api/index.go#L30

Reply:
I see - will try to avoid this then.

Reply:
done in 6cfb62773187466371a65ceb5c76f756110e6352